src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRazorPageFactoryProvider.cs

```diff
@@ -74,18 +74,19 @@ public RazorPageFactoryResult CreateFactory(string relativePath)
         /// <summary>
         /// Creates a factory for <see cref="IRazorPage"/>.
         /// </summary>
-        /// <param name="compiledType">The <see cref="Type"/> to produce an instance of <see cref="IRazorPage"/>
+        /// <param name="pageCreator">The <see cref="Func{Object}"/> to produce an instance of <see cref="IRazorPage"/>
```

Comment:
A `ViewLocationCacheItem` already contains a `Func<IRazorPage> PageFactory` property and `RazorPageFactoryResult` contains a similar `RazorPageFactory` property. May be caching similar things at two different levels i.e. before and after setting the `Path` property on the page. We should step back a bit e.g. create a single `Func<string, IRazorPage>` built using Linq expressions.